src/shared/components/enhancedReactTable/EnhancedReactTable.tsx

```diff
@@ -153,25 +168,24 @@ export default class EnhancedReactTable<T> extends React.Component<IEnhancedReac
         } = this.props;
 
         columns = columns || {};
-        let visibleCols:IColumnDefMap = this.resolveVisible(columns, this.state.columnVisibility);
+        let visibleCols:IColumnDefMap = this.visibleColsSelector(this.state, this.props);
 
         // dynamic reactable props (depends on the columnVisibility state)
-        let sortable:Array<string|IColumnSort> = this.resolveSortable(visibleCols);
-        let filterable:Array<string|IColumnFilter> = this.resolveFilterable(visibleCols);
+        let sortable:Array<string|IColumnSort> = this.sortableColsSelector(this.state, this.props);
+        let filterable:Array<string|IColumnFilter> = this.filterableColsSelector(this.state, this.props);
 
         // sort columns
-        let sortedCols:Array<IEnhancedReactTableColumnDef> = this.resolveOrder(visibleCols);
+        let sortedCols:Array<IEnhancedReactTableColumnDef> = this.sortedColsSelector(this.state, this.props);
 
         // always use the initially sorted columns (this.sortedColumns),
         // otherwise already hidden columns will never appear in the dropdown menu!
-        let columnVisibility:Array<IColumnVisibilityDef> = this.resolveColumnVisibility(
-            this.colNameToId, this.sortedColumns, this.state.columnVisibility);
+        let columnVisibility:Array<IColumnVisibilityDef> = this.columnVisibilityArraySelector(this.state, this.props);
 
         // column headers: an array of Th components
-        const headers = this.generateHeaders(sortedCols);
+        const headers = this.headersSelector(this.state, this.props);
 
         // table rows: an array of Tr components
-        const rows = this.generateRows(sortedCols, rawData);
+        const rows = this.rowsSelector(this.state, this.props);
```

Comment:
Lets discuss whether it makes sense to do this or just to build row collection as part of the constructor (instead of in render).   Both strategies would accomplish the same thing.  Just worth talking through.